### PR TITLE
Translate directly from Cirq to C++ circuit.

### PR DESCRIPTION
**Pull request by 95-martin-orion**
This should cover all of the gates required by #134. It also explicitly deprecates the file-input format; although we do not currently plan to remove file input, we will no longer maintain it. Python users are encouraged to use Cirq; C++ users will need to interface directly with the qsim libraries.

@sergeisakov: please confirm that this covers all of the `gates_cirq` gates (known exception: `MatrixGate`). If there are any other docs that need updating, let me know.

Follow-up work (will open new issues for these):
- MatrixGates
- Built-in decomposition of unknown gates

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
Of note: I deleted `qsim_test.py` since that file was specifically for testing the "call qsim from Python without Cirq` use case, which is no longer supported. (Calling qsim directly from C++ is still valid, though)

File: qsimcirq/qsim_circuit.py [hunk 1]
```diff
@@ -15,6 +15,82 @@
...
 
 import cirq
+from qsimcirq import qsim
+
+
+def _cirq_gate_kind(gate):
+  if isinstance(gate, cirq.ops.identity.IdentityGate):
+    if gate.num_qubits == 1:
+      return qsim.kI
+    return qsim.kI2
```
Comment (sergeisakov) [inline qsimcirq/qsim_circuit.py:25]: Check if `if gate.num_qubits == 2` for `qsim.kI2`.

File: qsimcirq/qsim_circuit.py [hunk 2]
```diff
@@ -15,6 +15,82 @@
...
+    return qsim.kPhasedISwapPowGate
+  if isinstance(gate, cirq.ops.FSimGate):
+    return qsim.kFSimGate
+  if isinstance(gate, cirq.ops.MatrixGate):
+    if gate.num_qubits == 1:
+      return qsim.kMatrixGate1
+    return qsim.kMatrixGate2
+  # No matches - decomposition is required.
+  # TODO: support decomposition.
+  return qsim.kGateDecomp
```
Comment (sergeisakov) [inline qsimcirq/qsim_circuit.py:93]: `kGateDecomp` is a gate from Schmidt decomposition. It is internal to qsim. If there are no matches, an error should be raised.

File: pybind_interface/pybind_main.cpp [hunk 3]
```diff
@@ -48,20 +48,12 @@ T parseOptions(const py::dict &options, const char *key) {
...
-  std::string circuit_str;
+Circuit<Cirq::GateCirq<float>> getCircuit(const py::dict &options) {
   try {
-    circuit_str = parseOptions<std::string>(options, "c\0");
+    return options["c\0"].cast<Circuit<Cirq::GateCirq<float>>>();
   } catch (const std::invalid_argument &exp) {
     throw;
   }
-  std::stringstream ss(circuit_str);
-  if (!CircuitQsimParser<IO>::FromStream(std::numeric_limits<unsigned>::max(),
```
Comment (sergeisakov) [inline pybind_interface/pybind_main.cpp:60]: `#include "../lib/circuit_qsim_parser.h"` above should be deleted.

File: qsimcirq/qsim_circuit.py (hunk 1 above)
Comment (95-martin-orion) [reply qsimcirq/qsim_circuit.py]: Done.

File: qsimcirq/qsim_circuit.py (hunk 2 above)
Comment (95-martin-orion) [reply qsimcirq/qsim_circuit.py]: Done.

File: pybind_interface/pybind_main.cpp (hunk 3 above)
Comment (95-martin-orion) [reply pybind_interface/pybind_main.cpp:60]: Done.

**Comment by 95-martin-orion** [comment]
Regarding missing checks: I think Github's "required checks" and the "only execute on C++ changes" rule for Bazel aren't cooperating. Options include (1) making Bazel checks optional, or (2) running Bazel checks even when no C++ changes are submitted.

I'm not thrilled about it, but since (1) leaves holes in our test coverage I'll send out a PR for (2). (This might be good incentive for me to revisit issue #69...)

**Pull request merged**